scripts/ci/pr-bot/processNewPrs.ts

```diff
@@ -0,0 +1,220 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const github = require("./shared/githubUtils");
+const { getChecksStatus } = require("./shared/checks");
+const commentStrings = require("./shared/commentStrings");
+const { ReviewerConfig } = require("./shared/reviewerConfig");
+const { PersistentState } = require("./shared/persistentState");
+const { Pr } = require("./shared/pr");
+const { REPO_OWNER, REPO, PATH_TO_CONFIG_FILE } = require("./shared/constants");
+import { CheckStatus } from "./shared/checks";
+
+// Returns true if the pr needs to be processed or false otherwise.
+// We don't need to process PRs that:
+// 1) Have WIP in their name
+// 2) Are less than 20 minutes old
+// 3) Are draft prs
+// 4) Are closed
+// 5) Have already been processed
+// 6) Have notifications stopped
+// 7) The pr doesn't contain the go label (temporary). TODO(damccorm) - remove this when we're ready to roll this out to everyone.
+// unless we're supposed to remind the user after tests pass
+// (in which case that's all we need to do).
+function needsProcessed(pull, prState: typeof Pr): boolean {
+  if (!pull.labels.find((label) => label.name.toLowerCase() == "go")) {
+    console.log(
+      `Skipping PR ${pull.number} because it doesn't contain the go label`
+    );
+    return false;
+  }
+  if (prState.remindAfterTestsPass && prState.remindAfterTestsPass.length > 0) {
```

Comment:
`prState.remindAfterTestsPass?.length > 0`

Reply:
That has the potential to coallese to `null > 0`, which I don't think is actually what we want.